packages/server/graphql/public/types/NewMeeting.ts

```diff
@@ -0,0 +1,69 @@
+import toTeamMemberId from '../../../../client/utils/relay/toTeamMemberId'
+import {getUserId} from '../../../utils/authorization'
+import isMeetingLocked from '../../types/helpers/isMeetingLocked'
+import {NewMeetingResolvers} from '../resolverTypes'
+
+const NewMeeting: NewMeetingResolvers = {
+  __resolveType: ({meetingType}) => {
+    const resolveTypeLookup = {
+      retrospective: 'RetrospectiveMeeting',
+      action: 'ActionMeeting',
+      poker: 'PokerMeeting',
+      teamPrompt: 'TeamPromptMeeting'
+    } as const
+    return resolveTypeLookup[meetingType as keyof typeof resolveTypeLookup]
+  },
+  createdByUser: ({createdBy}, _args, {dataLoader}) => {
+    return dataLoader.get('users').loadNonNull(createdBy)
+  },
+  facilitator: ({facilitatorUserId, teamId}, _args, {dataLoader}) => {
+    const teamMemberId = toTeamMemberId(teamId, facilitatorUserId)
+    return dataLoader.get('teamMembers').load(teamMemberId)
+  },
+  locked: async ({endedAt, teamId}, _args, {authToken, dataLoader}) => {
+    const viewerId = getUserId(authToken)
+    return isMeetingLocked(viewerId, teamId, endedAt, dataLoader)
+  },
+
+  meetingMembers: ({id: meetingId}, _args, {dataLoader}) => {
+    return dataLoader.get('meetingMembersByMeetingId').load(meetingId)
+  },
+  organization: async ({teamId}, _args, {dataLoader}) => {
+    const team = await dataLoader.get('teams').loadNonNull(teamId)
+    const {orgId} = team
+    return dataLoader.get('organizations').load(orgId)
+  },
+  phases: async ({phases, id: meetingId, teamId, endedAt}, _args, {authToken, dataLoader}) => {
+    const viewerId = getUserId(authToken)
+    const locked = await isMeetingLocked(viewerId, teamId, endedAt, dataLoader)
+
+    const resolvedPhases = phases.map((phase: any) => ({
+      ...phase,
+      meetingId,
+      teamId
+    }))
+
+    if (locked) {
+      // make all stages non-navigable so even if the user removes the overlay they cannot see all meeting data
+      return resolvedPhases.map((phase: any) => ({
+        ...phase,
+        stages: phase.stages.map((stage: any) => ({
```

Comment:
+1 this isn't new code, but the three `any`s could be removed here as they're inferred